### PR TITLE
Bump deps, build with Go 1.22.2

### PR DESCRIPTION
**Pull request by benashz**
Build with Go 1.22.2

Update deps manually with (auto dep upgrades workflow seems broken for this repo):
```
go list -u -m -json all | jq -r 'select(.Indirect != true and .Update != null) | .Path+"@"+.Update.Version' | grep -v 'armon/go-metrics' | xargs -L1 go get
```

Other fixes:
- build: fix generate target "no go files found" issue
- build: stop bitbucketing make target command output
- declare interface implementations
- add CODEOWNERS

Clean up:
- build: drop unused VETARGS
- sort and group local imports with: 

```
goimports -local "$(head -n1 go.mod | cut -d " " -f 2)" -w .
```

### CODE REVIEW

**Review by fairclothjm — APPROVED** [review]
LGTM

File: CHANGELOG.md
Comment (fairclothjm) [inline CHANGELOG.md:1]: Should we go ahead and add the heading for `v0.18.0`?

Comment (benashz) [reply CHANGELOG.md:1]: 👍

Comment (benashz) [reply CHANGELOG.md:1]: Updated in 477d7fe54961e7e248d7ce0dcac0bbdba9c0998d

**Review by fairclothjm — APPROVED** [review]

**Pull request merged**